dash/include/dash/algorithm/Generate.h

```diff
@@ -46,6 +46,8 @@ void generate (
   auto llast  = lrange.end;
 
   std::generate(lfirst, llast, gen);
+
+  first.team().barrier();
```

Comment:
Ok, I have no idea ... We should document that in the algorithm section. A quick survey showed that some other algorithms (e.g. [fill](https://github.com/dash-project/dash/blob/f716024fe3d8a5408dbaa2551ba1c8e0153fab42/dash/include/dash/algorithm/Fill.h#L35)) do not use barriers.